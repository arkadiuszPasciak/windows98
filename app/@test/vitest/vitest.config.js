import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
