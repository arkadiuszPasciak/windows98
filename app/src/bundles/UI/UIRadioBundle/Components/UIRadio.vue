<template>
  <label
    class="UIRadio"
    :class="disabled ? 'is-disabled' : ''"
  >
    <input
      class="input"
      type="radio"
      :value="modelValue"
      :disabled="disabled"
      :checked="checked"
      :name="name"
      :data-test="generateDataTestName('input', name, modelValue)"
      @input="$emit('update:modelValue', modelValue)"
    >

    <span
      class="radio"
      :data-test="generateDataTestName('icon', name, modelValue)"
    />

    <UIText
      v-if="$slots.default"
      class="text"
      :data-test="generateDataTestName('label', name, modelValue)"
    >
      <slot />
    </UIText>
  </label>
</template>

<script setup lang="ts">
import { generateDataTestName } from "@APP/src/bundles/UI/UIRadioBundle/Services/UIRadio.services"
import UIText from "@APP|Bundles/UITextBundle/Components/UIText.vue"

defineProps({
	modelValue: {
		type: String,
		required: true,
	},
	name: {
		type: String,
		required: true,
	},
	disabled: {
		type: Boolean,
		default: false,
	},
	checked: {
		type: Boolean,
		default: false,
	},
})

defineEmits(["update:modelValue"])
</script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/UIRadioBundle/Styles/UIRadio.styles.scss"
/>
