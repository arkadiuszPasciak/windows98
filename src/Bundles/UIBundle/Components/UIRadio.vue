<template>
  <label class="UIRadio" :class="`variant-${variant}`">
    <input
      class="input"
      type="radio"
      :name="name"
      :disabled="disabled"
      :checked="checked"
    />

    <span class="radio"></span>

    <p v-if="$slots.default" class="text"><slot /></p>
  </label>
</template>

<script setup lang="ts">
  import { PropType } from 'vue'

  defineProps({
    variant: {
      type: String as PropType<'primary'>,
      default: 'primary',
    },
    name: {
      type: String,
      required: true,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    checked: {
      type: Boolean,
      default: false,
    },
  })
</script>

<style lang="scss" scoped>
  @import '@Bundles/UIBundle/Styles/UIRadio.styles.scss';
</style>
