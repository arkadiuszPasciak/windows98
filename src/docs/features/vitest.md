# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
