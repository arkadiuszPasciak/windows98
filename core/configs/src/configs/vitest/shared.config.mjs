import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
