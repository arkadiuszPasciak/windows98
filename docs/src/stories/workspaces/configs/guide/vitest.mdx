import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

