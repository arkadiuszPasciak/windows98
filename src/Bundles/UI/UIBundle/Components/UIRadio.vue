<template>
  <label class="UIRadio">
    <input
      class="input"
      type="radio"
      :value="modelValue"
      :disabled="disabled"
      :checked="checked"
      :name="name"
      :data-test="generateDataTestName('input', name, modelValue)"
      @input="$emit('update:modelValue', modelValue)"
    />

    <span
      class="radio"
      :data-test="generateDataTestName('icon', name, modelValue)"
    ></span>

    <UIText
      v-if="$slots.default"
      class="text"
      :data-test="generateDataTestName('label', name, modelValue)"
    >
      <slot />
    </UIText>
  </label>
</template>

<script setup lang="ts">
  import UIText from '@Bundles/UIBundle/Components/UIText.vue'
  import { generateDataTestName } from '@Bundles/UIBundle/Services/UIRadio.services'

  defineProps({
    modelValue: {
      type: String,
      required: true,
    },
    name: {
      type: String,
      required: true,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    checked: {
      type: Boolean,
      default: false,
    },
  })

  defineEmits(['update:modelValue'])
</script>

<style lang="scss" scoped>
  @import '@Bundles/UIBundle/Styles/UIRadio.styles.scss';
</style>
