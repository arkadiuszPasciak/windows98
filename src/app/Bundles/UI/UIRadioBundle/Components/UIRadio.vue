<template>
  <label class="UIRadio">
    <input
      class="input"
      type="radio"
      :value="modelValue"
      :disabled="disabled"
      :checked="checked"
      :name="name"
      :data-test="generateDataTestName('input', name, modelValue)"
      @input="$emit('update:modelValue', modelValue)"
    />

    <span
      class="radio"
      :data-test="generateDataTestName('icon', name, modelValue)"
    ></span>

    <UIText
      v-if="$slots.default"
      class="text"
      :data-test="generateDataTestName('label', name, modelValue)"
    >
      <slot />
    </UIText>
  </label>
</template>

<script setup lang="ts">
  import UIText from '@APP|Bundles/UITextBundle/Components/UIText.vue'
  import { generateDataTestName } from '@APP|Bundles/UIRadioBundle/Services/UIRadio.services'

  defineProps({
    modelValue: {
      type: String,
      required: true,
    },
    name: {
      type: String,
      required: true,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    checked: {
      type: Boolean,
      default: false,
    },
  })

  defineEmits(['update:modelValue'])
</script>

<style
  lang="scss"
  scoped
  src="@APP|Bundles/UIRadioBundle/Styles/UIRadio.styles.scss"
/>
