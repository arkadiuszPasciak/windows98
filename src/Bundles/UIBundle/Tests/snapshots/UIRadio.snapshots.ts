import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import UIRadio from '@Bundles/UIBundle/Components/UIRadio.vue'

const availableWrapper = mount(UIRadio, {
  props: {
    variant: 'primary',
    disabled: false,
    checked: false,
    name: 'available-radio',
    modelValue: 'restart',
  },
  slots: {
    default: 'Check Me!',
  },
})

const unAvailableWrapper = mount(UIRadio, {
  props: {
    variant: 'primary',
    disabled: true,
    checked: true,
    name: 'unavailable-radio',
    modelValue: 'shut-down',
  },
})

describe('[UIBundle]<Snapshots>(UIRadio)', async () => {
  it('renders correctly component: variant primary, disabled false and checked false', () => {
    expect(availableWrapper.element).toMatchSnapshot()
  })

  it('renders correctly component: variant primary, disabled true and checked true', () => {
    expect(unAvailableWrapper.element).toMatchSnapshot()
  })
})
