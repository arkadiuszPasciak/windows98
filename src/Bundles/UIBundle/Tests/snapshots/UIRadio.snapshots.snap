// Vitest Snapshot v1

exports[`[UIBundle]<Snapshots>(UIRadio) > renders correctly component: variant primary, disabled false and checked false 1`] = `
<label
  class="UIRadio variant-primary"
  data-v-ccc35d9d=""
>
  <input
    class="input"
    data-v-ccc35d9d=""
    name="available-radio"
    type="radio"
  />
  <span
    class="radio"
    data-v-ccc35d9d=""
  />
  <p
    class="text"
    data-v-ccc35d9d=""
  >
    
    Check Me!
    
  </p>
</label>
`;

exports[`[UIBundle]<Snapshots>(UIRadio) > renders correctly component: variant primary, disabled true and checked true 1`] = `
<label
  class="UIRadio variant-primary"
  data-v-ccc35d9d=""
>
  <input
    class="input"
    data-v-ccc35d9d=""
    disabled=""
    name="unavailable-radio"
    type="radio"
  />
  <span
    class="radio"
    data-v-ccc35d9d=""
  />
  <!--v-if-->
</label>
`;
